tests/support/test.tcl

```diff
@@ -126,11 +126,11 @@ proc wait_for_condition {maxtries delay e _else_ elsescript} {
     }
 }
 
-# try to match a value to a list of patterns that is either regex, or plain sub-string
-proc search_pattern_list {value pattern_list {substr false}} {
+# try to match a value to a list of patterns that are either regex (starts with "/") or plain string.
+proc search_pattern_list {value pattern_list} {
     set n 0
     foreach el $pattern_list {
-        if {[string length $el] > 0 && ((!$substr && [regexp -- $el $value]) || ($substr && [string match $el $value]))} {
+        if {[string length $el] > 0 && (([string match /* $el] && [regexp -- $el /$value]) || [string equal $el $value])} {
```

Comment:
i don't think prefixing the value with `/` is good idea.
what if the regex starts with `$` (beginning of a line).
i think that instead we must trim the `/` from the expression

Reply:
fixed in `bf599dc558f99ad96ef253c1bdd4d530a14b0b17`